### PR TITLE
First version

### PR DESCRIPTION
**Pull request by mvalahtv**
# Install host dependency
```
ansible-playbook -vv -i mvala1, local.yml --tags install 
```
# Install upstream image (catalog)
```
ansible-playbook -vv -i mvala1, local.yml  \
-e run_upstream=true \
--tags catalogue_build
```

# Test operator
```
ansible-playbook -vv -i mvala1, local-test-operator.yml  \
-e run_upstream=true -e run_prereqs=false -e run_imagesource=false \
-e operator_dir="/tmp/community-operators-for-catalog/upstream-community-operators/aqua"
```

### CODE REVIEW

File: deploy-olm-operator.yml [hunk 1]
```diff
@@ -32,7 +32,7 @@
       until: login_result.rc == 0
       environment:
         KUBECONFIG: "{{ kubeconfig_path }}"
-      when: openshift_url is defined and ((distro_type is undefined) or (distro_type != "upstream"))
+      when: not run_upstream|bool
```
Comment (J0zi) [inline deploy-olm-operator.yml]: missed original `openshift_url` condition. Please add.

File: local-test-operator.yml [hunk 2]
```diff
@@ -17,26 +18,31 @@
...
+
+  pre_tasks:
+    - setup:
+      tags:
+        - always
+      when: run_upstream|bool
+
   tasks:
     - set_fact:
-        distro_type: "{{ DISTRO_TYPE | default('') }}"
```
Comment (J0zi) [inline local-test-operator.yml:24]: will be missing in roles?

File: local.yml [hunk 3]
```diff
@@ -54,14 +50,32 @@
...
-    - { role: todo, tags: ["pure_test"]}
+    # - { role: todo, tags: ["pure_test"]}
     # - { role: operator_courier_verify, tags: ["pure_test", "operator_courier_verify"] }
-    
+
   tasks:
-  
-    - name: debug
+
+    # - name: Running as user
```
Comment (J0zi) [inline local.yml:58]: needed?

**Comment by J0zi** [comment]
Please add a command to description

**Comment by J0zi** [comment]
it ends and not continue towards OLM and cleanup. Please have a look and enable lines which are enabled by default. I used:
ansible-playbook -vv -i operator-ansible,  -e run_upstream=true -e run_prereqs=true -e run_imagesource=true local.yml \
    -e kubeconfig_path="${KUBECONFIG_PATH}" \
    -e quay_namespace="'op-test'" \
    -e production_quay_namespace="community-operators" \
    -e operator_dir="/tmp/community-operators-for-catalog/upstream-community-operators/aqua" \
    -e run_imagesource=false


Result tail:

TASK [build_upstream_catalog : Create 'test-operators' namespace (it may fail, if already exists)] *********************************************************************************************************
task path: /home/jbreza/projects/RH/playbooks/mvalahtv/operator-test-playbooks/roles/build_upstream_catalog/tasks/main.yml:40
changed: [operator-ansible] => {"changed": true, "cmd": "kubectl create namespace test-operators", "delta": "0:00:00.079143", "end": "2020-05-21 14:00:52.013914", "rc": 0, "start": "2020-05-21 14:00:51.934771", "stderr": "", "stderr_lines": [], "stdout": "namespace/test-operators created", "stdout_lines": ["namespace/test-operators created"]}

TASK [Run distro_type] *************************************************************************************************************************************************************************************
task path: /home/jbreza/projects/RH/playbooks/mvalahtv/operator-test-playbooks/local.yml:64
ok: [operator-ansible] => {
    "msg": "distro_type : upstream"
}

TASK [Run kubernetes] **************************************************************************************************************************************************************************************
task path: /home/jbreza/projects/RH/playbooks/mvalahtv/operator-test-playbooks/local.yml:71
ok: [operator-ansible] => {
    "msg": "run_upstream : True"
}
META: ran handlers
META: ran handlers

PLAY RECAP *************************************************************************************************************************************************************************************************
operator-ansible           : ok=39   changed=19   unreachable=0    failed=0    skipped=10   rescued=0    ignored=3

File: local-test-operator.yml (hunk 2 above)
Comment (mvalawk) [reply local-test-operator.yml:24]: This is not needed, because `distro_type` is variable and not set as fact

File: local.yml (hunk 3 above)
Comment (mvalawk) [reply local.yml:58]: Not needed. We will do cleanup tomorrow

**Comment by mvalahtv** [comment]
For now use `locat-test-operator` for testing. `local.yaml` is used only for installationa and build upstream catalog image.

**Comment by mvalahtv** [comment]
I am running
```
ansible-playbook -vv -i mvala1, -e run_upstream=true -e run_prereqs=true -e run_imagesource=true local-test-operator.yml  -e kubeconfig_path="${KUBECONFIG_PATH}" -e quay_namespace="'op-test'" -e production_quay_namespace="community-operators" -e operator_dir="/tmp/community-operators-for-catalog/upstream-community-operators/aqua"
```

**Pull request merged**